### ISSUE TITLE
Loss of marks A and B after waking up from sleep

### ISSUE THREAD

**Issue by rustequal** [clsid2/mpc-hc#1578]
After the computer wakes up from sleep the player performs a video reload. But at the same time, pre-set marks A and B for video repeat are lost. It would be great if the player didn't lose these marks after sleep. I have to reconfigure these marks every time after sleep. I suspect this is due to a new change:

"The player now reloads the video file when resumed after hibernate or very long pause (> 30 min)"

Windows 10 Pro (x64, Version 21H1), MPC-HC 1.9.18 (64 bit)

**Comment by kkdexz** [comment]
Have you tried just disabling that setting? Options -> Advanced -> ReloadAfterLongPause => 0 (Default is 30). It is set in minutes. This will disable it and make it so it doesn't reload at all (or alternatively you can set/increase it to w/e amount you want, e.g. 60, 120). This will retain the A-B marks. Unless you actually do need it to reload after sleep. Then again, you can also reload it manually yourself, only after you have confirmed where the A-B marks are, so you can re-set them.

But yes, I do think it would be great if the player did this automatically on its own. Perhaps according to an additional entry in Options -> Player -> History -> "Remember A-B marks". I doubt this will be implemented tho as clsid2 supposedly isn't accepting any feature additions/requests.

**Comment by rustequal** [comment]
If I set this parameter to 0, then the reload still occurs after waking up and the marks are lost. I tried different values, and came to the conclusion that this "ReloadAfterLongPause" setting does not affect the behavior of the player after waking up. The player always reloads the video after waking up, regardless of the parameter "ReloadAfterLongPause" value.

**Comment by clsid2** [comment]
It currently always reloads in case of hibernate/sleep.

AB-repeat may get saved in the recent file history in the near future. But it is low priority for me.

**Comment by clsid2** [comment]
Next version will have option to disable reload after hibernate.

**Comment by rustequal** [comment]
> Next version will have option to disable reload after hibernate.

Thanks

**Comment by adipose** [comment]
I have this working pretty well, but I have noticed that DVD A-B repeat is pretty broken.  So I'm working on that.

**Comment by adipose** [comment]
https://github.com/clsid2/mpc-hc/pull/1655

Right now it expects file/dvd position saving to be enabled, or else it won't recover A-B repeat.  Not sure if we need a new flag for that.  One thing is, resuming might not work if it has to reload and lose the A-B repeat?  Does resuming work even if saving position is disabled?
